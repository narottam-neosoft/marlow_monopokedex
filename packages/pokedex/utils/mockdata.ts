export const mockStoreData = {
    pokemon: {
      allData: [],
      data: [
        {
          name: "bulbasaur",
          url: "https://pokeapi.co/api/v2/pokemon/1/",
        },
        { name: "ivysaur", url: "https://pokeapi.co/api/v2/pokemon/2/" },
      ],
      count: 0,
      pagination: {
        page: 0,
        pageSize: 10,
      },
    },
    pokemonDetail: {
      name: "Pikachu",
      height: "100",
      weight: "250",
      location_area_encounters: "https://pokeapi.co/api/v2/pokemon/3/encounters",
    },
  };