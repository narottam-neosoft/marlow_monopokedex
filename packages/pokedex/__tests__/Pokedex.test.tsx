import * as React from "react";
import Home from "../pages/index";
import "@testing-library/jest-dom";
import { screen, render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { mockStoreData } from "../utils/mockdata";

jest.mock("next/router", () => require("next-router-mock"));

const mockStore = configureStore([thunk]);
const store = mockStore(mockStoreData);
describe("pokemon-list", () => {
  it("List rendered successfully", () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("navigates to the Pokemon details page when a row is clicked", () => {
    const pushMock = jest.fn();
    const useRouterMock = jest.spyOn(require("next/router"), "useRouter");
    useRouterMock.mockImplementation(() => ({
      push: pushMock,
    }));

    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    fireEvent.click(screen.getByText("bulbasaur"));
    expect(pushMock).toHaveBeenCalledWith("/bulbasaur");
  });
});
