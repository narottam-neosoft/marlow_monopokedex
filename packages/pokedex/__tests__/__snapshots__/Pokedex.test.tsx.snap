// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-list List rendered successfully 1`] = `
<div>
  <div>
    <div
      style="height: 80vh; width: 100%;"
    >
      <div
        aria-colcount="2"
        aria-multiselectable="false"
        aria-rowcount="3"
        class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1687jxh-MuiDataGrid-root"
        role="grid"
      >
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
              role="rowgroup"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="1"
                class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                role="row"
              >
                <div
                  aria-colindex="1"
                  aria-label="Name"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor super-app-theme--header"
                  data-field="name"
                  role="columnheader"
                  style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="0"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Name
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-controls=":r2:"
                        aria-haspopup="true"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        id=":r3:"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-label="Url"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                  data-field="url"
                  role="columnheader"
                  style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Url
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-controls=":r6:"
                        aria-haspopup="true"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        id=":r7:"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDataGrid-virtualScroller css-axafay-MuiDataGrid-virtualScroller"
            style="overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 104px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="bulbasaur"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 52px; min-height: 52px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="0"
                    data-field="name"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="bulbasaur"
                    >
                      bulbasaur
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="1"
                    data-field="url"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="https://pokeapi.co/api/v2/pokemon/1/"
                    >
                      https://pokeapi.co/api/v2/pokemon/1/
                    </div>
                  </div>
                </div>
                <div
                  aria-rowindex="3"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                  data-id="ivysaur"
                  data-rowindex="1"
                  role="row"
                  style="max-height: 52px; min-height: 52px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="0"
                    data-field="name"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="ivysaur"
                    >
                      ivysaur
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="1"
                    data-field="url"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="https://pokeapi.co/api/v2/pokemon/2/"
                    >
                      https://pokeapi.co/api/v2/pokemon/2/
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
        >
          <div />
          <div
            class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                0–0 of 0
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
